inc/hal/adc_dma.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file adc_dma.hpp
+ *
+ * @brief ADC DMA HAL header
+ *
+ * @date 03/2024
+ */
+
+#ifndef __HAL_ADC_DMA_HPP__
+#define __HAL_ADC_DMA_HPP__
+
+#include <adc.h>
+#include <cstdint>
+#include <functional>
+
+namespace hal {
+/**
+ * @brief Class to handle ADC DMA peripheral on STM32 microcontrollers
+ */
+class AdcDma {
+    public:
+        /**
+         * @brief Configuration structure for ADC DMA
+         */
+        struct Config {
+            ADC_HandleTypeDef* handle;
+            void               (* init_function)(void);
+        };
```

Comment:
Pq que deixou a init_functoin desse jeito? Se for manter asssim pode tirar o include do functional

Reply:
É que o std::function tem bastante overhead, ela é uma classe bem complexa e pesada, mas nesse caso a gente só precisa de um ponteiro pra função mesmo, ai como já tem coisa da ST junto que é em C, não vejo problema deixar o ponteiro, ai tem q tirar o functional mesmo.